server/models/competition_instance.js

```diff
@@ -70,21 +90,24 @@ class CompetitionInstance {
     return this.hasStarted() ? this.comp.getWinners() : null;
   }
 
-  _createNewGame(gameInfo) {
+  _createNewGame(gameInfo, lastGame) {
     if (gameInfo) {
-      let gameId = this.gameEngine.create(gameInfo.gameParams);
-      let game = this.currentGame = this.gameEngine.get(gameId);
+      let lastGameState = lastGame ? { lastGame: _.omit(lastGame.game, "params") } : {};
```

Comment:
Correct, in a competition this would create a huge params object for each game, and the UI does not handle this properly (nor the database)

Besides, the competition sends the initial ```params``` to every game in the competition, so storing them this way is unnecessary